File: src/services/link-service/folder.mapper.ts
import { LinkMapper } from './link.mapper';
import { IFolder } from './types/folder.type';

export class FolderMapper {

    public static map(response: any): Array<IFolder> {
        return response.map((folder: any) => FolderMapper.mapSingle(folder));
    }

    public static mapSingle(folder: any): IFolder {
        return {
            identifier: folder.Identifier,
            name: folder.Name,
            icon: folder.Icon,
            sortOrder: folder.SortOrder,
            columnId: folder.ColumnId,
            links: folder.Links.map((link: any) => LinkMapper.mapSingle(link))
        };
    }

    public static mapToApi(folders: Array<IFolder>): any {
        return folders.map((folder: IFolder) => FolderMapper.mapToApiSingle(folder));
    }

    public static mapToApiSingle(folder: IFolder): any {
        return {
            Identifier: folder.identifier,
            Name: folder.name,
            Icon: folder.icon,
            SortOrder: folder.sortOrder,
            ColumnId: folder.columnId,
            Links: LinkMapper.mapToApi(folder.links)
        };
    }
}

File: api/home-box-landing/HomeBoxLanding.Api/Features/Folders/FolderMapper.cs
using HomeBoxLanding.Api.Features.Folders.Types;
using HomeBoxLanding.Api.Features.Links;

namespace HomeBoxLanding.Api.Features.Folders;

public class FolderMapper
{
    public static Folder Map(FolderRecord record)
    {
        return new Folder
        {
            Identifier = record.Identifier,
            Name = record.Name,
            Icon = record.Icon,
            SortOrder = record.SortOrder,
            ColumnId = record.ColumnIdentifier,
            Links = record.Links
                .OrderBy(x => x.SortOrder)
                .Select(LinkMapper.Map)
                .ToList()
        };
    }
}
